io/eolearn/io/sentinelhub_process.py

```diff
@@ -140,14 +141,14 @@ def _get_size(self, bbox: BBox) -> Tuple[int, int]:
         raise ValueError("Size or resolution for the requests should be provided!")
 
     @staticmethod
-    def _extract_bbox(bbox: Optional[BBox], eopatch: EOPatch) -> BBox:
-        if eopatch.bbox is None:
+    def _extract_bbox(bbox: Optional[BBox], eopatch_bbox: Optional[BBox]) -> BBox:
+        if eopatch_bbox is None:
             if bbox is None:
                 raise ValueError("Either the eopatch or the task must provide valid bbox.")
             return bbox
 
-        if bbox is None or eopatch.bbox == bbox:
-            return eopatch.bbox
+        if bbox is None or eopatch_bbox == bbox:
+            return eopatch_bbox
         raise ValueError("Either the eopatch or the task must provide bbox, or they must be the same.")
```

Comment:
Also, the name of the function should probably be updated, since it's not extracting anything anymore per se. Perhaps `_bbox_selector` or `_consolidate_bbox`. It could even be used just as a check which raises an error or passes through, perhaps that would be more elegant, since it would then be used as a util function in `_get_modified_eopatch` in `io/eolearn/io/extra/meteoblue.py`, where the same logic us used.

Reply:
best of luck convincing `mypy` that your reasoning is correct ;)
It should simplify itself a bit later, when eopatch.bbox wont ever be None.

Check functions do not provide any type information to the outside :/ so i'm afraid that is not that useful. I do like `_consolidate_bbox` though :D

Reply:
renamed